R/compengine.R

```diff
@@ -0,0 +1,780 @@
+
+#' CompEngine feature set
+#'
+#' Calculate the features that have been used in CompEngine database, using method introduced in package
+#' \code{kctsa}. 
+#' 
+#' The features involved can be grouped as \code{autocorrelation}, 
+#' \code{prediction}, \code{stationarity}, \code{distribution}, and \code{scaling}. 
+#'
+#' @param y the input time series
+#' @return a vector with CompEngine features
+#' @seealso \code{\link{hctsa_autocorr}}
+#' @seealso \code{\link{hctsa_pred}}
+#' @seealso \code{\link{hctsa_station}}
+#' @seealso \code{\link{hctsa_dist}}
+#' @seealso \code{\link{hctsa_scal}}
+#' @references B.D. Fulcher and N.S. Jones. hctsa: A computational framework for automated time-series phenotyping using massive feature extraction. Cell Systems 5, 527 (2017).
+#' @references B.D. Fulcher, M.A. Little, N.S. Jones Highly comparative time-series analysis: the empirical structure of time series and their methods. J. Roy. Soc. Interface 10, 83 (2013).
+#' @author Yangzhuoran Yang
+#' @export
+compengine <- function(x){
+  c(hctsa_autocorr(x), hctsa_pred(x), hctsa_station(x), hctsa_dist(x), hctsa_scal(x))
+}
+
+
+#' The autocorrelation feature set from software package \code{hctsa}
+#'
+#' Calculate the features that grouped as autocorrelation set, 
+#' which have been used in CompEngine database, using method introduced in package \code{kctsa}. 
+#' 
+#' Features in this set are \code{CO_Embed2_Basic_tau_incircle_1}, 
+#' \code{CO_Embed2_Basic_tau_incircle_2}, 
+#' \code{AC_9}, 
+#' \code{CO_FirstMin_ac}, 
+#' \code{CO_trev_1_num}, 
+#' \code{SB_MotifTwo_mean_hhh}, 
+#' and \code{PH_Walker_prop_01_sw_propcross}.
+#'
+#' @param y the input time series
+#' @return a vector with autocorrelation features
+#' @seealso \code{\link{CO_Embed2_Basic_tau_incircle}}
+#' @seealso \code{\link{AC_9}}
+#' @seealso \code{\link{CO_FirstMin_ac}}
+#' @seealso \code{\link{CO_trev_1_num}}
+#' @seealso \code{\link{SB_MotifTwo_mean_hhh}}
+#' @seealso \code{\link{PH_Walker_prop_01_sw_propcross}}
+#' @references B.D. Fulcher and N.S. Jones. hctsa: A computational framework for automated time-series phenotyping using massive feature extraction. Cell Systems 5, 527 (2017).
+#' @references B.D. Fulcher, M.A. Little, N.S. Jones Highly comparative time-series analysis: the empirical structure of time series and their methods. J. Roy. Soc. Interface 10, 83 (2013).
+#' @author Yangzhuoran Yang
+#' @export
+hctsa_autocorr <- function(x){
```

Comment:
This involves calling acf() many times which is not efficient. Also, we already have our own acf features function. Perhaps these can be combined?